hbase_sql/src/main/java/com/splicemachine/olap/OlapServerSubmitter.java

```diff
@@ -747,4 +745,22 @@ private Path getQualifiedLocalPath(URI localURI, Configuration hadoopConf) throw
         }
         return new Path(qualifiedURI);
     }
+
+    class RestartWatcher implements Watcher {
+        @Override
+        public void process(WatchedEvent watchedEvent) {
+            try {
+                if (watchedEvent.getType().equals(Event.EventType.NodeDataChanged)) {
+                    SpliceLogUtils.info(LOG, "Node data has changed");
+                    restarted = true;
+                    String root = HConfiguration.getConfiguration().getSpliceRootPath()
+                            + HBaseConfiguration.OLAP_SERVER_PATH + HBaseConfiguration.OLAP_SERVER_RESTART_PATH;
+                    RecoverableZooKeeper rzk = ZkUtils.getRecoverableZooKeeper();
+                    rzk.getData(root, new RestartWatcher(), null);
+                }
```

Comment:
Same here, we need to handle session events

Reply:
done